Wikipedia table scrapes

mustafaascha opened the pull request:
Hello! I added information about how to retrieve information from Wikipedia. I look forward to hearing from you.

mustafaascha committed:
Added wikipedia scrape

Added information about scraping tables from wikipedia

mustafaascha committed:
removed stringr reference

removed reference to stringr package, table was pretty from the getgo.

mustafaascha committed:
removed duplicate code at end

I removed duplicate code that I thought would display the table of cat names. I'm not sure how to display the table...it's showing up when I knitr the code in RStudio.

sckott commented:
@mustafaascha Thanks for this!  I have some comments I'll make inline on your changes

If you are okay using `httr` instead of `RCurl` we can get the table like `readHTMLTable(content(GET(url)))[[7]]`, and we don't have to specifically disable https

sckott commented on index.Rmd:
````diff
@@ -487,6 +488,58 @@ getdoge(c("wow", "suchhostility", "bemoreinclusive"))
+
+Fortunately, the RCurl package has a function that lets you choose not to use https. Phew! Here, we'll tell R to copy the webpage into a new object, "wikiCatNamePage". 
+```{r makePage, collapse=TRUE}
+wikiCatNamePage <- getURL(url,.opts = list(ssl.verifypeer = FALSE) )
````
You're right that this works to disable SSL verify peer, but I think for this basic R intro website, we can make this easier for people. In addition, I understand that it is not great to teach people to do this as it should not always be done. `httr` can handle this automatically. See example in main PR comments

sckott commented on index.Rmd:
````diff
@@ -487,6 +488,58 @@ getdoge(c("wow", "suchhostility", "bemoreinclusive"))
+First, we'll have to load some helpful packages.
+```{r scrapePackages, collapse=TRUE}
+library(XML)
+library(RCurl)
````
Can you use `httr` instead?  It is more actively maintained, and is easier to use

sckott commented on index.Rmd:
````diff
@@ -487,6 +488,58 @@ getdoge(c("wow", "suchhostility", "bemoreinclusive"))
+url <- "https://en.wikipedia.org/wiki/Popular_cat_names"
+```
+
+Wait a minute! This website looks like it uses https! That's a bit annoying, but we can work around it. If we want to scrape data from an https website, we'll have to tell R not to use the extra layer of security.
````
I think https is a good thing on the web in general, giving people anonymity while browsing. You're right that it can make scraping a bit harder. Can you change `... annoying` to `... annoying for scraping` or similar

sckott commented on index.Rmd:
```diff
@@ -487,6 +488,58 @@ getdoge(c("wow", "suchhostility", "bemoreinclusive"))
+
+It looks like the function stored seven different things in the object, "wikiCatNameTables". Looking at the webpage, I like the table from the subsection called "Cultural references to the naming of cats". One of my uncles is related to the son-in-law of the cousin of the co-creator of Tom, from Tom and Jerry. I know--I'm cool. 
+
+If we use the str() function to look at the objects in wikiCatNameTables, we'll see that the last item in the list has the data we want. The str() function shows too much information because of how wikiCatNameTables is organized, so the output isn't shown here. You should try it, though!
```
wrap `str()` in back-ticks as it's code, not general text

sckott commented on index.Rmd:
````diff
@@ -487,6 +488,58 @@ getdoge(c("wow", "suchhostility", "bemoreinclusive"))
+parsedWikiCatNamePage <- htmlParse(wikiCatNamePage)
+```
+
+Alright! Now we've organized the website so that a different function, htmlTables(), will be able to pull the tables that we want. 
````
wrap `htmlTables()` in back-ticks as it's code, not general text

sckott commented on index.Rmd:
````diff
@@ -487,6 +488,58 @@ getdoge(c("wow", "suchhostility", "bemoreinclusive"))
+famousCats
+```
+
+Wow! We did it! And the data looks great! Give yourself a pat on the back!
````
Can  you add a sentence suggesting that readers look at the package `rvest` for scraping?

mustafaascha committed:
edits

I replaced Rcurl with httr--it definitely simplified the code. I also added reference to rvest at the end, along with a suggestion to explore it. Finally, I wrapped the code with backticks.

Thanks for letting me participate!

mustafaascha committed:
backticked package name

I backticked the name of the package, "rvest".

sckott commented:
@mustafaascha very sorry, i forgot about this!  

looking now

sckott commented:
I'll do a few small changes after pulling down, but LGTM

mustafaascha commented:
Hi! Thanks so much!

This was my first time participating in anything on GitHub, it makes me
really happy to have contributed.

Take care!

On Thu, Oct 29, 2015 at 4:29 PM, Scott Chamberlain <notifications@github.com

> wrote:
> 
> Merged #15 https://github.com/sckott/rforcats/pull/15.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/sckott/rforcats/pull/15#event-449704424.

sckott commented:
thank you for the contribution :)
